src/internal/operations/backup.go

```diff
@@ -117,9 +117,65 @@ func (op BackupOperation) validate() error {
 // pointer wrapping the values, while those values
 // get populated asynchronously.
 type backupStats struct {
-	k             *kopia.BackupStats
-	ctrl          *data.CollectionStats
-	resourceCount int
+	k                   *kopia.BackupStats
+	ctrl                *data.CollectionStats
+	resourceCount       int
+	hasNewDetailEntries bool
+}
+
+// To qualify as an assist backup, all of the following must be true:
+// 1. new detail entries were produced during this operation
+// 2. The updated details file was persisted without error
+// 3. we have a valid snapshot ID
+// 4. we don't have any non-recoverable errors
+// 5. we recorded recoverable errors
+// 6. We are not running in best effort mode. Reason being that there is
+// no way to distinguish assist backups from merge backups in best effort mode
+// Primary reason for persisting assist backup models is to ensure we don't
+// lose corso extension data(deets) for items which were downloaded and
+// processed by kopia during this backup operation.
+// Note: kopia.DetailsMergeInfoer doesn't impact decision making for creating
+// assist backups. It may be empty if it’s the very first backup so there is no
+// merge base to source base details from, or non-empty, if there was a merge base.
+// In summary, if there are no new deets, no new extension data was produced
+// and hence no need to persist assist backup model.
+func isAssistBackup(
+	newDeetsProduced bool,
+	snapID, ssid string,
+	failurePolicy control.FailurePolicy,
+	err *fault.Bus,
+) bool {
+	return newDeetsProduced &&
+		snapID != "" &&
+		ssid != "" &&
+		err.Failure() == nil &&
+		len(err.Recovered()) > 0 &&
+		failurePolicy != control.BestEffort
+}
+
+// To qualify as a merge backup, all of the following must be true:
+// 1. we have a valid snapshot ID
+// 2. valid ssid, details were persisted without error
+// 3. we don't have any non-recoverable errors
+// 4. no recoverable errors if not running in best effort mode
+func isMergeBackup(
+	snapID, ssid string,
+	failurePolicy control.FailurePolicy,
+	err *fault.Bus,
+) bool {
+	if snapID == "" || ssid == "" {
+		return false
+	}
+
+	if err.Failure() != nil {
+		return false
+	}
+
+	if failurePolicy == control.BestEffort {
+		return true
+	}
```

Comment:
Honestly, I think we can remove this. Incremental base selection is a black box to end users and it's not really safe to have backups that could have errors in the merge base set. We don't expect many people to be using this anyway
```suggestion
```

Reply:
Great! Sounds good

Reply:
Do note that after removing BestEffort checks in `isMergeBackup` &`isAssistBackup`, here is the new behavior for users who may continue using BestEffort mode.

1. If user runs into recoverable errors, we will persist an assist backup.
2. Say there is a file which always fails to download with a recoverable error, we will never be able to persist a merge backup for this user. 

Given above, i think we should just remove support for BestEffort mode during backup. I'll create a ticket for this.

Reply:
>we will never be able to persist a merge backup for this user. So no restore possibility.

I'm not certain this is the case right now. The backup list logic and restore logic don't check tags as far as I know, they just check for the existence of backup models. If they only check if the backup model exists, has an item data snapshot and a backup details snapshot then a restore will technically be possible

Reply:
That makes sense. Should we also add a PR to only report merge backups during `backup list`?
Returning assist backups in that list seems wrong, since assist backups are an intermediate state with recoverable errors, and we should probably prevent restoring from them?

Reply:
yea, there should be a PR to update how list works. As far as what we filter out, I'm still debating that a little. As you've pointed out, we should probably technically display best effort backups as completed since they well, completed. Unfortunately this may put us at odds between tags and display :/ Open to discussion on this point, and the discussion doesn't necessarily have to happen in this thread

Reply:
Sounds good. I propose the following:

1.  We keep the best effort mode checks in `isMergeBackup` & `isAssistBackup`. So we will continue to persist merge backups for best effort modes. 
2. We add changes to `list backup` to only return merge backups i.e. they are not tagged with `AssistBackup` tag. 
3. Regarding future of best effort mode, I'm creating an internal doc for discussion.

Does this sound good to you? :)

Reply:
yup, sounds good!